peer/msg_router_test.go

```diff
@@ -0,0 +1,143 @@
+package peer
+
+import (
+	"testing"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type mockEndpoint struct {
+	mock.Mock
+}
+
+func (m *mockEndpoint) Name() string {
+	args := m.Called()
+
+	return args.String(0)
+}
+
+func (m *mockEndpoint) CanRouteTo(msg lnwire.Message) bool {
+	args := m.Called(msg)
+
+	return args.Bool(0)
+}
+
+func (m *mockEndpoint) SendMessage(msg lnwire.Message) bool {
+	args := m.Called(msg)
+
+	return args.Bool(0)
+}
+
+// TestMessageRouterOperation tests the basic operation of the message router:
+// add new endpoints, route to them, remove, them, etc.
+func TestMessageRouterOperation(t *testing.T) {
+	msgRouter := NewPeerMsgRouter()
+	msgRouter.Start()
+	defer msgRouter.Stop()
+
+	openChanMsg := &lnwire.OpenChannel{}
+	commitSigMsg := &lnwire.CommitSig{}
+
+	errorMsg := &lnwire.Error{}
+
+	// For this test, we'll have two endpoints, each with distinct names.
+	// One endpoint will only handle OpenChannel, while the other will
+	// handle the CommitSig message.
+	fundingEndpoint := &mockEndpoint{}
+	fundingEndpointName := "funding"
+	fundingEndpoint.On("Name").Return(fundingEndpointName)
+	fundingEndpoint.On("CanRouteTo", openChanMsg).Return(true)
+	fundingEndpoint.On("CanRouteTo", errorMsg).Return(false)
+	fundingEndpoint.On("CanRouteTo", commitSigMsg).Return(false)
+	fundingEndpoint.On("SendMessage", openChanMsg).Return(true)
+
+	commitEndpoint := &mockEndpoint{}
+	commitEndpointName := "commit"
+	commitEndpoint.On("Name").Return(commitEndpointName)
+	commitEndpoint.On("CanRouteTo", commitSigMsg).Return(true)
+	commitEndpoint.On("CanRouteTo", openChanMsg).Return(false)
+	commitEndpoint.On("CanRouteTo", errorMsg).Return(false)
+	commitEndpoint.On("SendMessage", commitSigMsg).Return(true)
```

Comment:
since we do allow 2 endpoints to handle the same message - let's test that case too?

Reply:
The 🐰 is way ahead of ya ;) 
https://github.com/lightningnetwork/lnd/pull/8434#discussion_r1468285987

Reply:
🤓 🐰

Reply:
I don't think we want multiple dispatch :)